genesyscloud/architect_grammar/resource_genesyscloud_architect_grammar_utils.go

```diff
@@ -0,0 +1,231 @@
+		}
+
+		languageList = append(languageList, languageMap)
+	}
+
+	return languageList
+}
+
+func flattenGrammarLanguageFileMetadata(d *schema.ResourceData, fileMetadata *platformclientv2.Grammarlanguagefilemetadata, languageCode string, fileType string) []interface{} {
+	if fileMetadata == nil {
+		return nil
+	}
+
+	metadataMap := make(map[string]interface{})
+
+	resourcedata.SetMapValueIfNotNil(metadataMap, "file_name", fileMetadata.FileName)
+	resourcedata.SetMapValueIfNotNil(metadataMap, "file_type", fileMetadata.FileType)
+	if schemaResource, ok := d.Get("languages").([]interface{}); ok {
+		for _, language := range schemaResource {
+			if languageMap, ok := language.(map[string]interface{}); ok {
+				if fmt.Sprintf("%v", languageMap["language"]) != languageCode {
+					continue
+				}
+				if fileType == "voice" {
+					if voiceData, ok := languageMap["voice_file_data"].([]interface{}); ok {
+						voiceDataMap := voiceData[0].(map[string]interface{})
+						if hash, ok := voiceDataMap["file_content_hash"].(string); ok {
+							metadataMap["file_content_hash"] = hash
+						}
+					}
+				} else if fileType == "dtmf" {
+					if dtmfData, ok := languageMap["dtmf_file_data"].([]interface{}); ok {
+						dtmfDataMap := dtmfData[0].(map[string]interface{})
+						if hash, ok := dtmfDataMap["file_content_hash"].(string); ok {
+							metadataMap["file_content_hash"] = hash
+						}
+					}
+				}
+			}
+		}
+	}
+
+	return []interface{}{metadataMap}
+}
+
+func ArchitectGrammarResolver(grammarId, exportDirectory, subDirectory string, configMap map[string]interface{}, meta interface{}) error {
+	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getArchitectGrammarProxy(sdkConfig)
+
+	fullPath := path.Join(exportDirectory, subDirectory)
+	if err := os.MkdirAll(fullPath, os.ModePerm); err != nil {
+		return err
+	}
+
+	ctx := context.Background()
+	grammar, _, err := proxy.getArchitectGrammarById(ctx, grammarId)
+	if err != nil {
+		return err
+	}
+
+	if grammar.Languages == nil {
+		return nil
+	}
+
+	for _, language := range *grammar.Languages {
+		if language.VoiceFileMetadata != nil && language.VoiceFileUrl != nil {
+			fileType := ""
+			if language.VoiceFileMetadata.FileType != nil {
+				fileType = strings.ToLower(*language.VoiceFileMetadata.FileType)
+			}
+			voiceFileName := fmt.Sprintf("%s-voice-%s.%s", *language.Language, grammarId, fileType)
+			if err := files.DownloadExportFile(fullPath, voiceFileName, *language.VoiceFileUrl); err != nil {
+				return err
+			}
+		}
+
+		if language.DtmfFileMetadata != nil && language.DtmfFileUrl != nil {
+			fileType := ""
+			if language.DtmfFileMetadata.FileType != nil {
+				fileType = strings.ToLower(*language.DtmfFileMetadata.FileType)
+			}
+			dtmfFileName := fmt.Sprintf("%s-dtmf-%s.%s", *language.Language, grammarId, fileType)
+			if err := files.DownloadExportFile(fullPath, dtmfFileName, *language.DtmfFileUrl); err != nil {
+				return err
+			}
+		}
```

Comment:
These two if blocks are repetitive. Could make a function out of it. Generics might come in handy